src/one/nio/serial/Repository.java

```diff
@@ -372,14 +372,13 @@ private static Serializer generateFor(Class<?> cls) {
                     serializer = new CollectionSerializer(cls);
                 } else if (Map.class.isAssignableFrom(cls) && !hasOptions(cls, FIELD_SERIALIZATION)) {
                     serializer = new MapSerializer(cls);
-                } else if (Serializable.class.isAssignableFrom(cls)) {
-                    if (cls.getName().startsWith("java.time.") && JavaInternals.findMethod(cls, "writeReplace") != null) {
+                } else {
+                    boolean extendsSerializable = Serializable.class.isAssignableFrom(cls);
+                    if (extendsSerializable && cls.getName().startsWith("java.time.") && JavaInternals.findMethod(cls, "writeReplace") != null) {
                         serializer = new JavaTimeSerializer(cls);
                     } else {
-                        serializer = new GeneratedSerializer(cls);
+                        serializer = new GeneratedSerializer(cls, !extendsSerializable);
                     }
-                } else {
-                    serializer = new InvalidSerializer(cls);
```

Comment:
Replacement of `InvalidSerializer` by `GeneratedSerializer` (JSON-only) looks like a serious behavior change, because `GeneratedSerializer` implements `Serializer` methods, but `InvalidSerializer` throws exceptions. So a user of the upstream version would get an exception if she forgot to `implement Serializable`, but in this version she would not.

Reply:
Yeah, I think as suggested by @atimofeyev maybe it makes sense to make a separete JsonSerializer with a bit different behavior. Such as not needing to implement Serializable.